lib/unknown-attributes-filter.coffee

```diff
@@ -0,0 +1,43 @@
+debug = require('debug')('sphere-product-import:unknown-attributes-filter')
+_ = require 'underscore'
+_.mixin require 'underscore-mixins'
+Promise = require 'bluebird'
+
+class UnknownAttributesFilter
+
+  constructor: (@logger) ->
+    debug "Unknown Attributes Filter initialized."
+
+  filter: (productType, product) =>
+    new Promise (resolve) =>
+      resolve() unless productType.attributes
+      attrNameList = _.pluck(productType.attributes, 'name')
+      Promise.all [
+        @_filterVariantAttributes(product.masterVariant, attrNameList),
+        Promise.map product.variants, (variant) =>
+          @_filterVariantAttributes(variant, attrNameList)
+        , {concurrency: 5}
+      ]
```

Comment:
Actually you just simply concat masterVariant + variants and map over that list.